### ISSUE TITLE
[Feature Request] Templates for logging/plotting

### ISSUE THREAD

**Issue by pkdogcom** [pytorch/ignite#64]
I think although right now the logging/plotting are straightforward to add using the event handler as in the examples, it can potentially be made even easier by adding some templates or some level of abstractions as logging/plotting are usually pretty standard and most people simply plot mostly the same types of loss, accuracy, learning rates, weights, etc. for the same type of task. 

For example, there can be a logging/plotting class which uses the low-level event handler API underlyingly but abstracts away implementation details like type of logger (file/tensorboard/wisdom), logging formating and provide standard implementation for certain common logging/ploting handlers like in the ignite examples or even task like classification/detection/regression by wrapping up logging/plotting handlers as a package, while still allows users to customize the behaviors of certain handlers or add extra handlers.

**Comment by alykhantejani** [comment]
@pkdogcom sounds like an interesting idea, what do you envisage the API looking like?

**Comment by pkdogcom** [comment]
I think the code snippets elanmart provides in #63 look like a good idea to me. And if the user wants to customize certain behaviors, they can either use the low-level event handler directly or subclass the template class and override certain types of logging/plotting inside.

**Comment by alykhantejani** [comment]
Agreed, this can also live in the contrib area

**Comment by Robin-des-Bois** [comment]
Is anyone working on this? If not I might give it a go, as I already have written some related code for myself.

**Comment by vfdev-5** [comment]
@Robin-des-Bois thanks for the proposition. Yes, I think you can work on this issue.
I think @pkdogcom talks about [`Timer`](https://github.com/pytorch/ignite/blob/master/ignite/handlers/timing.py#L9). 

I think a template for logging with Tensorboard can be useful, e.g. metrics. The difficulty is that it can be rather difficult to cover a generic data. 

Anyway let's discuss what can we add to ignite to simpify the workflow.

**Comment by anmolsjoshi** [comment]
Closing this in favor of #455
